hail/python/hail/expr/types.py

```diff
@@ -2282,18 +2281,12 @@ def is_primitive(t) -> bool:
 
 @typecheck(t=HailType)
 def is_container(t) -> bool:
-    return isinstance(t, tarray) or isinstance(t, tset) or isinstance(t, tdict)
+    return isinstance(t, (tarray, tdict, tset))
 
 
 @typecheck(t=HailType)
 def is_compound(t) -> bool:
-    return (
-        is_container(t)
-        or isinstance(t, tstruct)
-        or isinstance(t, tunion)
-        or isinstance(t, ttuple)
-        or isinstance(t, tndarray)
-    )
+    return isinstance(t, (tndarray, tstruct, ttuple, tunion))
```

Comment:
okay, it might only make this change under `--unsafe-fixes`, in which case i can also go back and just do safe fixes for now to make sure we don't introduce anything else incorrect by accident

Reply:
Ah, yes I would prefer not using `--unsafe-fixes`, at least at first

Reply:
😮 apparently it is a ruff bug, this happens without `--unsafe-fixes` on too!

Reply:
what are your thoughts on proceeding forward with this? i can fix this specific instance of the bug, but should we wait to run ruff on the hail directory til it's fixed, in case it's changing things we might have missed, or do you think manual review of everything is doable?